calyx-opt/src/passes/compile_static.rs

```diff
@@ -75,99 +87,265 @@ fn make_guard_dyn(
 }
 
 // Takes in static assignment `assign` and returns a dynamic assignments
-// Mainly does two things:
-// 1) if `assign` writes to a go/done hole, we should change that to the go/done
-// hole of the new, dynamic group instead of the old static group
-// 2) for each static Info guard (e.g. %[2:3]), we need to convert that to
-// dynamic guards, using `fsm`.
-// E.g.: %[2:3] gets turned into fsm.out >= 2 & fsm.out <= 3
+// Mainly transforms the guards such that fsm.out >= 2 & fsm.out <= 3
 fn make_assign_dyn(
     assign: ir::Assignment<StaticTiming>,
-    dyn_group: &ir::RRC<ir::Group>,
     fsm: &ir::RRC<ir::Cell>,
     fsm_size: u64,
     builder: &mut ir::Builder,
 ) -> ir::Assignment<Nothing> {
-    let new_dst = if assign.dst.borrow().is_hole() {
-        // holes should be either go/done
-        if assign.dst.borrow().name == "go" {
-            dyn_group.borrow().get("go")
-        } else {
-            panic!("hole port other than go port")
-        }
-    } else {
-        // if dst is not a hole, then we should keep it as is for the new assignment
-        assign.dst
-    };
     ir::Assignment {
         src: assign.src,
-        dst: new_dst,
+        dst: assign.dst,
```

Comment:
Often times when you're replacing some fields of a struct, you can use the following syntax:
```
ir::Assignment {
  guard: make_guard_dyn(...)
  ..assign
}
```

The `..assign` tells rust to replace all the other fields. This might not work here because you're building an `ir::Assignment<Nothing>` here from a `ir::Assignment<StaticTiming>`.

Reply:
yeah, doesn't work cuz of type differences unfortunately.